packages/uauth/src/index.ts

```diff
@@ -0,0 +1,336 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import type { StaticJsonRpcProvider as StaticJsonRpcProviderType } from '@ethersproject/providers'
+import UAuth from '@uauth/js'
+import type { UserInfo } from '@uauth/js'
+import {
+  WalletInit,
+  createEIP1193Provider,
+  EIP1193Provider,
+  Chain,
+  ProviderAccounts
+} from '@web3-onboard/common'
+
+interface uauthOptions {
+  clientID: string
+  redirectUri: string
+  scope?: string
+  shouldLoginWithRedirect?: boolean
+  bridge?: string
+  qrcodeModalOptions?: {
+    mobileLinks: string[]
+  }
+  connectFirstChainId?: boolean
+}
+
+export async function getUauthUser(options: uauthOptions) {
+  const { clientID, redirectUri, scope = 'openid wallet' } = options || {}
+
+  const uauthInstance = new UAuth({
+    clientID: clientID,
+    redirectUri: redirectUri,
+    scope: scope
+  })
+
+  return await uauthInstance.user().then(user => {
+    return user
+  })
+}
+
+function uauth(options: uauthOptions): WalletInit {
+  return () => {
```

Comment:
Will we need validation for the required uAuth options?

Reply:
For this the clientID and redirectUri are both required options with the rest being optional. If they aren't present, the uauth library will handle the errors when the Unstoppable login is selected. See screenshot. When putting this together I did try throwing errors if they weren't present but it caused severe breakage in the "Connect Wallet" flow of the demo app by blocking all login options. 
Very open to thought on this though!
![Screenshot 2022-11-15 at 1 41 41 PM](https://user-images.githubusercontent.com/28935974/202031117-5247efb2-dc13-4d98-8550-47f83be02319.png)

Reply:
It might be helpful to validate prior to this. We use Joi for most of our validation - an example can be seen here - https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/magic/src/validation.ts
or
https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/core/src/validation.ts

Reply:
Added validation according to the above